tensorflow_quantum/core/ops/tfq_utility_ops.py

```diff
@@ -58,3 +58,36 @@ def padded_to_ragged2d(masked_state):
     col_mask = tf.abs(tf.cast(masked_state[:, 0], tf.float32)) < 1.1
     masked = tf.ragged.boolean_mask(masked_state, col_mask)
     return padded_to_ragged(masked)
+
+
+def resolve_parameters(programs, symbol_names, symbol_values):
+    """Replace symbols in a batch of programs with concrete values.
+
```

Comment:
Can we also document the fact that this has the ability to "partially resolve" given parameters and therefore does not require you provide resolution values for ALL symbols in the circuit. I feel like a note that this op is NOT differentiable would also help our users understand that this op has very specific use cases.

Reply:
Expanded docstring